wire-swift-generator/src/main/java/com/squareup/wire/swift/SwiftGenerator.kt

```diff
@@ -449,19 +471,31 @@ class SwiftGenerator private constructor(
         .addParameter("from", reader, protoReader)
         .throws(true)
         .apply {
-          // Declare locals into which everything is writen before promoting to members.
+          // Declare locals into which everything is written before promoting to members.
           type.fields.forEach { field ->
-            val localType = if (field.isRepeated || field.isMap) {
-              field.typeName
-            } else {
-              field.typeName.makeOptional()
-            }
-            val initializer = when {
-              field.isMap -> "[:]"
-              field.isRepeated -> "[]"
-              else -> "nil"
+            when (type.syntax) {
+              PROTO_2 -> {
+                val localType = if (field.isRepeated || field.isMap) {
```

Comment:
Maybe make a `proto2InitialValue` ?

Reply:
I made this change.  It's now the only place where `proto3InitialValue` is read, so we could inline that if it would be more understandable